shell/models/fleet.cattle.io.gitrepo.js

```diff
@@ -325,6 +329,37 @@ export default class GitRepo extends SteveModel {
     return bds.filter((bd) => bd.metadata?.labels?.['fleet.cattle.io/repo-name'] === this.name);
   }
 
+  get allBundlesStatuses() {
+    const { nonReadyResources, ...bundlesSummary } = this.status?.summary || {};
```

Comment:
Is it correct to use GitRepo `status` to calculate Bundles `statuses` ?
I have an active bundle associated to my `test` GitRepo in my test environment but the UI shows `0/0`, I think this could be one of the root causes.

GitRepo `test`

![image](https://github.com/user-attachments/assets/ca894264-87e0-4915-8b69-39647982d603)

Dashboard page - 'Ready Bundles' column is `0/0`:

![image](https://github.com/user-attachments/assets/b2b002b3-47ad-491d-8fa3-e8438b974786)

Reply:
Yes, GitRepo's `[status.summary](https://github.com/rancher/fleet/blob/main/pkg/apis/fleet.cattle.io/v1alpha1/status.go#L5)` is indeed a `[BundleSummary](https://github.com/rancher/fleet/blob/main/pkg/apis/fleet.cattle.io/v1alpha1/bundle_types.go#L247)` type, which aggregates the status of all the Bundles that were deployed.

From the screenshots, I can see that you are using just a Local cluster with no downstreams configured, while creating a GitRepo within the `fleet-default` workspace, which excludes the local cluster. In that situation, it makes sense that the number of matching clusters and bundles is zero.

Reply:
I was wrong, despite its name, `BundleSummary` contains `BundleDeployment` counts, so this won't work.
I reworked it to obtain the Bundles statuses from BundleDeployments directly (still avoiding to use `Bundles` directly just for this, since they are expensive to download/process).